.github/workflows/goreleaser.yml

```diff
@@ -0,0 +1,31 @@
+on:
```

Comment:
I have copied it from https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/main/.github/workflows/release.yml

I don't think it is necessary to set this "generate_token" step, as this seems to be a holdover from the CZI implementation. 

I am also not able to test this github action locally except with: https://github.com/nektos/act. However, we would need to set any secrets as environment variables and I have no idea what our GPG_PRIVATE_KEY or PASSPHRASE is anymore. 

Since I don't care that much about this, I have reverted it back to what it used to be